src/Symfony/EventListener/WriteListener.php

```diff
@@ -110,6 +110,7 @@ public function onKernelView(ViewEvent $event): void
                 'uri_variables' => $uriVariables,
                 'resource_class' => $operation->getClass(),
                 'previous_data' => false === $operation->canRead() ? null : $request->attributes->get('previous_data'),
+                'data' => false === $operation->canRead() ? null : $request->attributes->get('data'),
```

Comment:
isn't `data` the same as `$controllerResult`?

Reply:
Yes, when using Symfony events, the `ReadListener` sets `data` in the request attributes. Then the `PayloadArgumentResolver` returns the value of `data`. This value is then passed as first argument to the `PlaceholderAction` which returns that value. This returned value is then set as `ViewEvent::$controllerResult`.